class.phpmailer.php

```diff
@@ -1474,8 +1496,8 @@ protected function smtpSend($header, $body)
         }
 
         // Attempt to send to all recipients
-        foreach (array($this->to, $this->cc, $this->bcc) as $togroup) {
-            foreach ($togroup as $to) {
+        if (!empty($this->envelope_to)) {
+            foreach ($this->envelope_to as $to) {
```

Comment:
Thank you for your replies.  That is correct and intended as per RFC.  Currently i am sick and will add an example when i am back again. My example would be an implementation snippet.

// Make mailer object using SMTP method
$mail = new PHPMailer;
$mail->isSMTP();
$mail->SMTPDebug = 2;

// Bypass actuall receipiant(s)
$mail->addEnvelopeTo('bart@vrancken.eu', 'Bart');

// Enforce these headers with their original values
foreach($headers['from'] as $from) {
    $mail->setFrom($from['address'], $from['name']);
}
foreach($headers['to'] as $to) {
    $mail->addAddress($to['address'], $to['name']);
}
$mail->Subject = !empty($headers['subject']) ? $headers['subject'] : '';
$mail->Body = !empty($body) ? $body : '';
$mail->MessageID = !empty($headers['message-id']) ? $headers['message-id'] : '';
$mail->MessageDate = !empty($headers['date']) ? $headers['date'] : date('D, j M Y H:i:s O');
$mail->ContentType = !empty($headers['content-type']) ? str_replace(';', '', $headers['content-type']) : '';
$mail->XMailer = 'AbuseIO Failed Mail Sender';

// Add headers accourding to RFC 2822 section 3.6.6.
$mail->addCustomHeader('Resent-From', $resendFrom);
$mail->addCustomHeader('Resent-To', $resendTo);
$mail->addCustomHeader('Resent-Date', $mail->buildMessageDate());
$mail->addCustomHeader('Resent-Message-Id', $mail->buildMessageID());
$mail->addCustomHeader('Resent-User-Agent', 'AbuseIO Failed Mail Sender');


if (!$mail->send()) {
    echo "Mailer Error: " . $mail->ErrorInfo;
} else {
    echo "Message sent!";
}